prow/kube/client.go

```diff
@@ -229,10 +233,17 @@ func (c *Client) doRequest(method, deckPath, urlPath, contentType string, query
 	}
 	var buf io.Reader
 	if body != nil {
-		b, err := json.Marshal(body)
-		if err != nil {
-			return nil, err
+		var b []byte
+		var merr error
```

Comment:
a) Looks like this can be defined inside the else block
b) Any reason not to use err here?